lib/miq_expression/field.rb

```diff
@@ -102,7 +102,7 @@ def column_type
     if custom_attribute_column?
       CustomAttribute.where(:name => custom_attribute_column_name, :resource_type => model.to_s).first.try(:value_type)
     else
-      target.type_for_attribute(column).type
+      target.type_for_attribute(column).type unless target.kind_of? String
```

Comment:
No longer needed? Perfect, because this change felt like some kind of hack. Removed it, so this PR is only about search filter.